src/main/scala/com/github/drmarjanovic/app/api/routes/package.scala

```diff
@@ -0,0 +1,18 @@
+package com.github.drmarjanovic.app.api
+
+import zhttp.http._
+import zio.Chunk
+
+import scala.util.Try
+
+package object routes {
+
+  def getQueryParam[A](req: Request, key: String): Option[A] =
+    req.url.queryParams.get(key).flatMap(_.headOption).flatMap(value => Try(value.asInstanceOf[A]).toOption)
```

Comment:
Specialize to numbers and use `toIntOption`. I'd also pack these as syntax extensions, e.g.:

```scala
implicit final class RequestOps(val req: Request) extends AnyVal {
  def limit: Option[Int] = ...

  def offset: Option[Int] = ...
}
```